src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,39 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        workWithNull(user);
+        if (user.getAge() < MIN_AGE) {
+            throw new RuntimeException("you are not of legal age");
+        }
+        if (user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new RuntimeException("Your password length shod be more or equals: "
+                    + MIN_PASSWORD_LENGTH);
+        }
+        return storageDao.add(user);
+    }
+
+    public void workWithNull(User user) {
```

Comment:
It is not really good to separate logic in this way. Better divide on check age, check login